symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/stream/OboStreamService.java

```diff
@@ -19,20 +25,49 @@
   /**
    * {@link StreamService#getStreamInfo(String)}
    *
-   * @param streamId    The stream id
+   * @param streamId    The stream id.
    * @return The information about the stream with the given id.
    * @see <a href="https://developers.symphony.com/restapi/reference#stream-info-v2">Stream Info V2</a>
    */
-  V2StreamAttributes getStreamInfo(String streamId);
+  V2StreamAttributes getStreamInfo(@Nonnull String streamId);
 
   /**
    * {@link StreamService#listStreams(StreamFilter)}
    *
-   * @param filter      The stream searching criteria
+   * @param filter      The stream searching criteria.
    * @return The list of streams retrieved according to the searching criteria.
    * @see <a href="https://developers.symphony.com/restapi/reference#list-user-streams">List Streams</a>
    */
-  List<StreamAttributes> listStreams(StreamFilter filter);
+  List<StreamAttributes> listStreams(@Nullable StreamFilter filter);
+
+  /**
+   * {@link StreamService#listStreams(StreamFilter, PaginationAttribute)}
+   *
+   * @param filter      The stream searching criteria.
+   * @param pagination  The skip and limit for pagination.
+   * @return The list of streams retrieved according to the searching criteria.
+   * @see <a href="https://developers.symphony.com/restapi/reference#list-user-streams">List Streams</a>
+   */
+  List<StreamAttributes> listStreams(@Nullable StreamFilter filter, @Nonnull PaginationAttribute pagination);
+
+  /**
+   * {@link StreamService#listAllStreams(StreamFilter)}
+   *
+   * @param filter The stream searching criteria.
+   * @return a {@link Stream} of matching streams according to the searching criteria.
+   * @see <a href="https://developers.symphony.com/restapi/reference#list-user-streams">List Streams</a>
+   */
+  Stream<StreamAttributes> listAllStreams(@Nullable StreamFilter filter);
```

Comment:
I'd feel more comfortable if we annotate those methods with `@API(status = EXPERIMENTAL)` (only the ones that return `Stream<>`)